src/Compilers/Test/Core/CompilationVerifier.cs

```diff
@@ -446,5 +446,25 @@ public void VerifySynthesizedFields(string containingTypeName, params string[] e
                 .ToList();
             AssertEx.SetEqual(expectedFields, members);
         }
+
+        /// <summary>
+        /// Useful for verifying the expected variables are hoisted for closures, async, and iterator methods.
+        /// Additionally this overload performs validations against the produced assembly.
+        /// </summary>
+        public void VerifySynthesizedFields(string containingTypeName, Func<IModuleSymbol, INamedTypeSymbol> findContainer, int arity, params string[] expectedFields)
+        {
+            VerifySynthesizedFields(containingTypeName, expectedFields);
+
+            CommonTestBase.RunValidators(this, assemblyValidator: null, symbolValidator: module =>
```

Comment:
>CommonTestBase.RunValidators(this, assemblyValidator: null, symbolValidator: module => [](http://example.com/codeflow?start=12&length=86)

What is the advantage of doing validation this way and dealing with ISymbols? it feels like we could share the validator function declared next to the affected tests and simply pass it to CompileAndVerify that is supposed to be called anyway. So far, I don't think adding a new helper into the shared layer is justified.

Reply:
OK, please check eecb17b.